client/js/libs/handlebars/parse.js

```diff
@@ -1,135 +1,140 @@
-        key: style[1],
-        keyregex: new RegExp(escaped + "(.*?)(" + escaped + "|$)")
-    };
-});
 
-function colors(line) {
-    // http://www.mirc.com/colors.html
-    // http://www.aviran.org/stripremove-irc-client-control-characters/
-    // https://github.com/perl6/mu/blob/master/examples/rules/Grammar-IRC.pm
-    // regexs are cruel to parse this thing
-
-    // already done?
-    if (!styleCheck_Re.test(line)) return line;
-
-    // split up by the irc style break character ^O
-    if (line.indexOf(styleBreak) >= 0) {
-        return line.split(styleBreak).map(colors).join("");
-    }
-
-    var result = line;
-    var parseArr = result.split(colourKey);
-    var text, match, colour, background = "";
-    for (var i = 0; i < parseArr.length; i++) {
-        text = parseArr[i];
-        match = text.match(back_re);
-        if (!match) {
-            // ^C (no colour) ending. Escape current colour and carry on
-            background = "";
-            continue;
-        }
-		colour = "irc-fg" + +match[1];
-		// set the background colour
-		if (match[3]) {
-			background = " irc-bg" + +match[3];
-		}
-        // update the parsed text result
-        result = result.replace(colourKey + text, styleTemplate({
-            style: colour + background,
-            text: text.slice(match[0].length)
-        }));
-    }
-
-    // Matching styles (italics/bold/underline)
-    // if only colours were this easy...
-    styles.forEach(function(style) {
-        if (result.indexOf(style.key) < 0) return;
-        result = result.replace(style.keyregex, function(match, text) {
-            return styleTemplate({
-                "style": style.style,
-                "text": text
-            });
-        });
-    });
-
-    return result;
+	/**
+	 * Parse mIRC formatting (bold, italic and underline)
+	 * https://github.com/myano/jenni/wiki/IRC-String-Formatting
+	 */
+	function parseFormatting(msg) {
+		return msg.replace(
+			/^(.*?)(\x02|\x1D|\x1F)([^\x0F\2]*)[\x0F\2]?(.*)$/,
+			function(match, before, format, formatted, after) {
+				console.log("Format: ", match, before, format, formatted, after);
+				processed = true;
+
+				var classes = {
+					"\x02": "irc-bold",
+					"\x1D": "irc-italic",
+					"\x1F": "irc-underline"
+				};
+
+				return parseMessage(before, parser+1)
+					+ "<span class='" + classes[format] + "'>"
+					+ parseMessage(formatted, parser)
+					+ "</span>"
+					+ parseMessage(after, parser);
+			}
+		);
+	}
+
+
+	/**
+	 * Parse mIRC color formatting
+	 * https://github.com/myano/jenni/wiki/IRC-String-Formatting
+	 */
+	function parseColors(msg) {
+		return msg.replace(
+			/^(.*?)\x03([0-9]{1,2}(?:,[0-9]{1,2})?)([^\x0F\x03]*)\x0F?(.*)$/,
+			function(match, before, color, formatted, after) {
+				console.log("Format: ", match, before, color, formatted, after);
+				processed = true;
+				color = color.split(",");
+
+				var classes = "irc-fg" + parseInt(color[0], 10);
```

Comment:
Actually it's needed for the cases where `\x0304color` is sent. We'd generate `<span class="irc-fg04">` which doesn't exist. So that makes sure it doesn't have a leading 0.